doc/platform_api/CMIS_Diagnostic_Monitoring_Overview_in_SONiC.md

```diff
@@ -0,0 +1,1685 @@
+- `TRANSCEIVER_VDM_LWARN_FLAG`: Stores the low warning flag for the VDM data.
+- `TRANSCEIVER_VDM_HALARM_FLAG_CHANGE_COUNT`: Keeps a count of how many times each VDM flag has changed for high alarm flags.
+- `TRANSCEIVER_VDM_LALARM_FLAG_CHANGE_COUNT`: Keeps a count of how many times each VDM flag has changed for low alarm flags.
+- `TRANSCEIVER_VDM_HWARN_FLAG_CHANGE_COUNT`: Keeps a count of how many times each VDM flag has changed for high warning flags.
+- `TRANSCEIVER_VDM_LWARN_FLAG_CHANGE_COUNT`: Keeps a count of how many times each VDM flag has changed for low warning flags.
+- `TRANSCEIVER_VDM_HALARM_SET_TIME`: Records the timestamp when each VDM high alarm flag was set.
+- `TRANSCEIVER_VDM_LALARM_SET_TIME`: Records the timestamp when each VDM low alarm flag was set.
+- `TRANSCEIVER_VDM_HWARN_SET_TIME`: Records the timestamp when each VDM high warning flag was set.
+- `TRANSCEIVER_VDM_LWARN_SET_TIME`: Records the timestamp when each VDM low warning flag was set.
+- `TRANSCEIVER_VDM_HALM_CLEAR_TIME`: Records the timestamp when each VDM high alarm flag was cleared.
+- `TRANSCEIVER_VDM_LALM_CLEAR_TIME`: Records the timestamp when each VDM low alarm flag was cleared.
+- `TRANSCEIVER_VDM_HWARN_CLEAR_TIME`: Records the timestamp when each VDM high warning flag was cleared.
+- `TRANSCEIVER_VDM_LWARN_CLEAR_TIME`: Records the timestamp when each VDM low warning flag was cleared.
+- `TRANSCEIVER_STATUS`: Stores the status data of the transceivers.
+- `TRANSCEIVER_STATUS_FLAG_CHANGE_COUNT`: Stores the count of times the transceiver status flag has changed.
+- `TRANSCEIVER_STATUS_FLAG_SET_TIME`: Records the timestamp when the transceiver status flag was set.
+- `TRANSCEIVER_STATUS_FLAG_CLEAR_TIME`: Records the timestamp when the transceiver status flag was cleared.
+- `TRANSCEIVER_PM`: Stores performance monitoring data.
+
+### 2.1 Transceiver DOM
+
+#### 2.1.1 Transceiver DOM sensor data
+
+The `TRANSCEIVER_DOM_SENSOR` table stores the real-time DOM data for the optical transceivers.
+
+lane_num: Represents the lane number of the field. The lane number is an integer value that ranges from 1 to 8.
+
+```plaintext
+    ; Defines Transceiver DOM sensor information for a port
+    key                          = TRANSCEIVER_DOM_SENSOR|ifname    ; information module DOM sensors on port
+    ; field                      = value
+    temperature                  = FLOAT                            ; temperature value in Celsius
+    voltage                      = FLOAT                            ; voltage value in V
+    tx{lane_num}power            = FLOAT                            ; tx power in dBm for each lane
+    rx{lane_num}power            = FLOAT                            ; rx power in dBm for each lane
+    tx{lane_num}bias             = FLOAT                            ; tx bias in mA for each lane
+    laser_temperature            = FLOAT                            ; laser temperature value in Celsius
```

Comment:
There could be more than one laser and more than one laser temperatures in one module. 
Laser temperature could be from 
1. aux2 or aux3 mon value from byte 20-21, or 22-23, In this case there will be only one laser temp value. 
2. VDM observable type 4, it may have more than one values representing lasers on different lanes.